Update httpclient-guidelines.md (#36866)
dotnet/docs: docs/fundamentals/networking/http/httpclient-guidelines.md

For more information about managing `HttpClient` lifetime with `IHttpClientFactory`, see [`IHttpClientFactory` guidelines](../../../core/extensions/httpclient-factory.md#httpclient-lifetime-management).

## Resilience policies with static clients

> It is possible to configure a `static` or *singleton* client to use any number of resilience policies using the following pattern:

Reviewer: ```suggestion
It's possible to configure a `static` or *singleton* client to use any number of resilience policies using the following pattern:
```

```csharp

> var retryPolicy = HttpPolicyExtensions

Reviewer: Where does the `HttpPolicyExtensions` type come from? We need both a NuGet package referenced here and a `using <namespace>`.

Author: The `HttpPolicyExtensions` comes from `Polly.Extensions.Http`

For more information about managing `HttpClient` lifetime with `IHttpClientFactory`, see [`IHttpClientFactory` guidelines](../../../core/extensions/httpclient-factory.md#httpclient-lifetime-management).

## Resilience policies with static clients

It's possible to configure a `static` or *singleton* client to use any number of resilience policies using the following pattern:

```csharp
var retryPolicy = HttpPolicyExtensions
    .HandleTransientHttpError()
    .WaitAndRetryAsync(3, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)));

var socketHandler = new SocketsHttpHandler { PooledConnectionLifetime = TimeSpan.FromMinutes(15) };
var pollyHandler = new PolicyHttpMessageHandler(retryPolicy)
{
    InnerHandler = socketHandler;
};

var httpClient = new HttpClient(pollyHandler);
```

> 

Reviewer: ```suggestion

The preceding code:

- Relies on [Microsoft.Extensions.Http.Polly](https://www.nuget.org/packages/Microsoft.Extensions.Http.Polly) NuGet package, transitively the [Polly.Extensions.Http](https://www.nuget.org/packages/Polly.Extensions.Http) NuGet package for the `HttpPolicyExtensions` type.
- Specifies a transient HTTP error handler, configured with retry policy that with each attempt will exponentially backoff delay intervals.
- Defines a pooled connection lifetime of fifteen minutes for the `socketHandler`.
- Passes the `socketHandler` to the `policyHandler` with the retry logic.
- Instantiates an `HttpClient` given the `policyHandler`.

```
